staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go

```diff
@@ -362,3 +364,16 @@ func isBackendPoolOnSameLB(newBackendPoolID string, existingBackendPools []strin
 
 	return true, "", nil
 }
+
+func isRunningAsKubelet() (bool, error) {
+	exe, err := osext.Executable()
+	if err != nil {
+		return false, fmt.Errorf("cloud not find the service executable: %v", err)
+	}
+
+	if strings.Contains(filepath.Base(exe), "kubelet") {
```

Comment:
what about hyperkube?

Reply:
this seems hacky in general though, wondering if there's a better way to do this

Reply:
Yep, this is a little hacky, any suggestions on this?

Reply:
Not sure yet, but probably warrants a follow-up.

Reply:
Yep, let me address this in a separate PR.